resolver/auth_mutations.resolvers_test.go

```diff
@@ -199,369 +120,288 @@ func prepareErrorFromConfig() struct {
-		name:    ErrorFromGenerateToken,
-		req:     req,
+func errorFromGenerateTokenCase() LoginType {
+	return LoginType{
+		name: ErrorFromGenerateToken,
+		req: loginArgs{
+			UserName: testutls.MockEmail,
+			Password: OldPassword,
+		},
 		wantErr: true,
 		err:     resultwrapper.ErrUnauthorized,
 	}
 }
-
-func prepareErrorUpdateUser() struct {
-	name     string
-	req      LoginArgs
-	wantResp *fm.LoginResponse
-	wantErr  bool
-	err      error
-} {
-	return struct {
-		name     string
-		req      LoginArgs
-		wantResp *fm.LoginResponse
-		wantErr  bool
-		err      error
-	}{
-		name:    ErrorUpdateUser,
-		req:     req,
+func errorUpdateUserCase() LoginType {
+	return LoginType{
+		name: ErrorUpdateUser,
+		req: loginArgs{
+			UserName: testutls.MockEmail,
+			Password: OldPassword,
+		},
 		wantErr: true,
 		err:     fmt.Errorf(ErrorMsgfromUpdateUser),
 	}
 }
-
-func prepareSuccessCase() struct {
-	name     string
-	req      LoginArgs
-	wantResp *fm.LoginResponse
-	wantErr  bool
-	err      error
-} {
-	return struct {
-		name     string
-		req      LoginArgs
-		wantResp *fm.LoginResponse
-		wantErr  bool
-		err      error
-	}{
+func loginSuccessCase() LoginType {
+	return LoginType{
 		name: SuccessCase,
-		req:  req,
+		req: loginArgs{
+			UserName: testutls.MockEmail,
+			Password: OldPassword,
+		},
 		wantResp: &fm.LoginResponse{
 			Token:        "jwttokenstring",
 			RefreshToken: TestToken,
 		},
 	}
 }
-
-func handleTestCases(t *testing.T, tt struct {
-	name     string
-	req      LoginArgs
-	wantResp *fm.LoginResponse
-	wantErr  bool
-	err      error
-}) {
-	// Prepare necessary mocks and patches
-	prepareMocksAndPatches(tt)
-
-	// Load environment variables
-	loadEnvironmentVariables()
-	// Handle specific test cases
-	handleSpecificTestCase(t, tt)
-}
-
-func prepareMocksAndPatches(tt struct {
-	name     string
-	req      LoginArgs
-	wantResp *fm.LoginResponse
-	wantErr  bool
-	err      error
-}) {
-	if tt.name == ErrorFromConfig {
-		patch := gomonkey.ApplyFunc(config.Load, func() (*config.Configuration, error) {
-			return nil, fmt.Errorf(ErrorMsgFromConfig)
-		})
-		defer patch.Reset()
+func loadLoginTestCases() []LoginType {
+	return []LoginType{
+		errorFindingUserCase(),
+		errorPasswordValidationCase(),
+		errorActiveStatusCase(),
+		errorFromConfigCase(),
+		errorFromJwtCase(),
+		errorFromGenerateTokenCase(),
+		errorUpdateUserCase(),
+		loginSuccessCase(),
 	}
+}
 
-	var tg jwt.Service
-	if tt.name == ErrorFromJwt {
-		patch := gomonkey.ApplyFunc(service.JWT, func(cfg *config.Configuration) (jwt.Service, error) {
-			return tg, fmt.Errorf(ErrorMsgFromJwt)
-		})
-		defer patch.Reset()
-	}
-	patch := gomonkey.ApplyFunc(tg.GenerateToken, func(u *models.User) (string, error) {
-		if tt.name == ErrorFromGenerateToken {
+func applyGenerateTokenPatchLogin(name string) *gomonkey.Patches {
+	tg := jwt.Service{}
+	return gomonkey.ApplyFunc(tg.GenerateToken, func(u *models.User) (string, error) {
+		if name == ErrorFromGenerateToken {
```

Comment:
this should be part of the init function. All logic associated to the test should be encapsulated in the init function